src/backend/joanie/core/factories.py

```diff
@@ -683,6 +684,22 @@ def main_invoice(self, create, extracted, **kwargs):
 
         return None
 
+    @factory.post_generation
+    # pylint: disable=method-hidden
+    def payment_schedule(self, create, extracted, **kwargs):
+        """
+        Cast input strings for the fields `amount` and `due_date` into the appropriate types
+        """
+        if create and extracted:
+            for item in extracted:
+                if isinstance(item["due_date"], str):
+                    item["due_date"] = date.fromisoformat(item["due_date"])
```

Comment:
WDYT about raising a proper exception if a bad string is provided ?
```suggestion
                    try:
                        item["due_date"] = date.fromisoformat(item["due_date"])
                    except ValueError as e:
                        raise OrderFactoryException(
                            f"Invalid date format for due_date: {e}"
                        )
```